src/prefect/cli/deploy.py

```diff
@@ -699,7 +716,17 @@ async def _run_multi_deploy(
                 app.console.print("Skipping unnamed deployment.", style="yellow")
                 continue
         app.console.print(Panel(f"Deploying {deploy_config['name']}", style="blue"))
-        await _run_single_deploy(deploy_config, actions, ci=ci)
+
+        await _run_single_deploy(
+            deploy_config,
+            actions,
+            ci=ci,
+            build_push_once=build_push_once,
+            image_has_been_built=image_has_been_built,
+        )
+
+        if build_push_once and not image_has_been_built:
+            image_has_been_built = True
```

Comment:
Right now I handle tracking if we should build the image once and if we have built it, outside of `_run_single_deploy`, as currently it looks is something like:

```python
def deploy():
    _run_multi_deploy()

def _run_multi_deploy():
    for d in deployment:
        _run_single_deploy()
```

so with the change if would look like this:
```python
def deploy():
    _run_multi_deploy()

def _run_multi_deploy():
    image_has_been_built = False
    for d in deployment:
        _run_single_deploy(build_push_once, image_has_been_built)
        if build_push_once and not image_has_been_built:
            image_has_been_built = True
```